build_tools/cmake/iree_bitcode_library.cmake

```diff
@@ -48,6 +48,9 @@ function(iree_bitcode_library)
     "-fno-ident"
     "-fdiscard-value-names"
 
+    # Limited debug information that is useful in profilers like Tracy.
+    "-gline-tables-only"
```

Comment:
Strange... build errors?

https://github.com/openxla/iree/actions/runs/7055395717/job/19206429344?pr=15756#step:7:426

```
[58/130] Generating /work/build-e2e-test-artifacts/e2e_test_artifacts/iree_module_PoseNet_fp32_tflite___x86_64-cascadelake-linux_gnu-llvm_cpu__experimental-flags_data-tiling_ukernel_/module.vmfb from PoseNet_fp32(tflite) [x86_64-cascadelake-linux_gnu-llvm_cpu][experimental-flags,data-tiling,ukernel]
FAILED: e2e_test_artifacts/iree_module_PoseNet_fp32_tflite___x86_64-cascadelake-linux_gnu-llvm_cpu__experimental-flags_data-tiling_ukernel_/module.vmfb /work/build-e2e-test-artifacts/e2e_test_artifacts/iree_module_PoseNet_fp32_tflite___x86_64-cascadelake-linux_gnu-llvm_cpu__experimental-flags_data-tiling_ukernel_/module.vmfb 
cd /work/build-e2e-test-artifacts/tests/e2e/test_artifacts && /work/full-build-dir/install/bin/iree-compile --output-format=vm-bytecode --mlir-print-op-on-diagnostic=false --iree-hal-target-backends=llvm-cpu --iree-input-type=tosa --iree-llvmcpu-target-triple=x86_64-unknown-linux-gnu --iree-llvmcpu-target-cpu=cascadelake --iree-opt-data-tiling --iree-llvmcpu-enable-ukernels=all /work/build-e2e-test-artifacts/e2e_test_artifacts/iree_PoseNet_fp32_tflite_.mlir -o /work/build-e2e-test-artifacts/e2e_test_artifacts/iree_module_PoseNet_fp32_tflite___x86_64-cascadelake-linux_gnu-llvm_cpu__experimental-flags_data-tiling_ukernel_/module.vmfb --iree-hal-executable-object-search-path=\"/work/build-e2e-test-artifacts\"
<unknown>:0: error: failed to run LLVM-IR opt passes for IREE::HAL::ExecutableOp targeting 'x86_64-unknown-unknown-eabi-elf'
<unknown>:0: error: failed to serialize executable for target backend llvm-cpu
<unknown>:0: error: failed to serialize executables
```

Reply:
Crazy, reproduces 100%:

```
tools/iree-compile --output-format=vm-bytecode --mlir-print-op-on-diagnostic=false --iree-hal-target-backends=llvm-cpu --iree-input-type=tosa --iree-llvmcpu-target-triple=x86_64-unknown-linux-gnu --iree-llvmcpu-target-cpu=cascadelake --iree-opt-data-tiling --iree-llvmcpu-enable-ukernels=all /home/benoit/iree-build/e2e_test_artifacts/iree_PoseNet_fp32_tflite_.mlir -o /tmp/a.vmfb
```

Reply:
It stops reproducing if I elide the constants in this model...!

Reply:
It stops reproducing if I just run `iree-opt` at all with no flags on this model!

Reply:
The failure occurs in module verification after LLVMIRPasses. One of the dispatch functions fails to verify. If I dump the IR at that point as `...failed.bc` and then run that through `llvm-dis`, I get:

```
!dbg attachment points at wrong subprogram for function
!151 = distinct !DISubprogram(name: "main_dispatch_5_mmt4d_1428x3x24x16x16x1_f32", linkageName: "main_dispatch_5_mmt4d_1428x3x24x16x16x1_f32", scope: !5, file: !5, line: 1, type: !72, scopeLine: 1, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !10)
ptr @main_dispatch_5_mmt4d_1428x3x24x16x16x1_f32
  %29 = shl i64 %28, 10, !dbg !152
!152 = !DILocation(line: 71, column: 55, scope: !153, inlinedAt: !157)
!160 = distinct !DISubprogram(name: "iree_uk_mmt4d", scope: !154, file: !154, line: 151, type: !155, scopeLine: 158, flags: DIFlagPrototyped | DIFlagAllCallsDescribed, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !67)
!160 = distinct !DISubprogram(name: "iree_uk_mmt4d", scope: !154, file: !154, line: 151, type: !155, scopeLine: 158, flags: DIFlagPrototyped | DIFlagAllCallsDescribed, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !67)
llvm-project/bin/llvm-dis: warning: ignoring invalid debug info in /tmp/x/module_llvm_module_linked_llvm_cpu_embedded_elf_x86_64.failed.bc
~
```

The failure point is here, https://github.com/llvm/llvm-project/blob/a456885efc95eb141d051ade517432c531d427f7/llvm/lib/IR/Verifier.cpp#L2966 , and if I dump SP and F in that failing `SP->describes(&F)` condition, I see that SP is the above !160/!161 describing `iree_uk_mmt4d` and F is the dispatch function `main_dispatch_5_mmt4d_1428x3x24x16x16x1_f32`.

So there is something wrong that happens in inlining the ukernel into the dispatch function.

That seems similar to what is tested in LLVMs own unit tests exercising this error:
https://github.com/llvm/llvm-project/blob/839a8fecb4c5dfe1b4484d5fc942a9490867c47a/llvm/test/Verifier/callsite-dbgloc.ll#L4

Should we debug this as an inlining bug in LLVM or, since we noted above that simply pre-processing this input model with `iree-opt` causes the error to disappear, add that `iree-opt` preprocessing to the importing of TFLite models?

@ScottTodd @benvanik @MaheshRavishankar

Reply:
I found that the problem goes away when passing `--iree-hal-dump-executable-sources-to`.

I debugged what it is exactly about this flag that removes the issue. It is this line: https://github.com/llvm/llvm-project/blob/29762e372294e5b591684f57979f33c1c48bbfba/mlir/lib/Transforms/LocationSnapshot.cpp#L58

After `translateModuleToLLVMIR` here, https://github.com/openxla/iree/blob/a28304473ad9526521510ad4e0a829c3c775c947/compiler/plugins/target/LLVMCPU/LLVMCPUTarget.cpp#L333
when things go well (i.e. when applying `--iree-hal-dump-executable-sources-to` or, as noted in above comments, when the input file has been run through `iree-opt`), the dispatch function has `!dbg` debug info metadata all over, not just on the function but on each op inside the function. But then things are not going well (when applying neither of the above workarounds, in which case we run into the error), the debug info is missing from the ops inside of the function.

Reply:
Oof this is weird. Not something we want spreading to other tests / models, but a few hacks in `build_tools/cmake/iree_benchmark_suite.cmake` are okay I guess...